### PR TITLE
Add set_static_field method

### LINKED ISSUE #226: Expose SetStatic*Field APIs
It looks like this crate exposes the `GetStatic*Field` APIs:
https://docs.rs/jni/0.14.0/jni/struct.JNIEnv.html#method.get_static_field

but not the corresponding `SetStatic*Field` APIs:
https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#wp20829

It would be nice to support those as well.

### PR DESCRIPTION
**Pull request by stanislav-tkach**
## Overview

Closes #226.

### Definition of Done

- [x] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has documentation
- [x] This change is not breaking **or** mentioned in the Changelog
- [ ] The [continuous integration build](https://www.travis-ci.org/jni-rs/jni-rs) passes

### CODE REVIEW

File: src/wrapper/jnienv.rs [hunk 1]
```diff
@@ -1693,51 +1693,47 @@ impl<'a> JNIEnv<'a> {
...
+                jni_unchecked!(self.internal, GetStaticDoubleField, class, field).into()
+            }
+            JP(Primitive::Byte) => {
+                jni_unchecked!(self.internal, GetStaticByteField, class, field).into()
+            }
+            JP(Primitive::Void) => {
+                return Err(ErrorKind::WrongJValueType("void", "see java field").into())
+            }
+        };
+        return Ok(result);
```
Comment (maurolacy) [inline src/wrapper/jnienv.rs:1736]: `Ok(result)` would work as well here.

File: src/wrapper/jnienv.rs [hunk 2]
```diff
@@ -1757,6 +1753,42 @@ impl<'a> JNIEnv<'a> {
         self.get_static_field_unchecked(class, (class, field, sig), ty)
     }
 
+    /// Set a static field. Requires a class lookup and a field id lookup internally.
+    pub fn set_static_field<'c, 'f, T, U>(&self, class: T, field: U, value: JValue) -> Result<()>
+    where
+        T: Desc<'a, JClass<'c>>,
+        U: Desc<'a, JStaticFieldID<'f>>,
+        //U: Into<JNIString>,
```
Comment (maurolacy) [inline src/wrapper/jnienv.rs:1761]: \<nit\> Remove commented code.

File: src/wrapper/jnienv.rs [hunk 3]
```diff
@@ -1757,6 +1753,42 @@ impl<'a> JNIEnv<'a> {
...
+    {
+        let class = class.lookup(self)?.into_inner();
+        let field = field.lookup(self)?.into_inner();
+
+        match value {
+            JValue::Object(v) => jni_unchecked!(
```
Comment (maurolacy) [inline src/wrapper/jnienv.rs:1767]: What happened to `JavaType::Array()`?

Comment (stanislav-tkach) [reply src/wrapper/jnienv.rs:1766]: As far as I can see, there is no way to do something like `SetStaticArrayField`. I'm not a Java expert, so I can be missing something.

**Comment by stanislav-tkach** [comment]
@maurolacy Thanks for the review! By the way, I think that pointing the the commented code isn't a nitpick. :+1:

File: src/wrapper/jnienv.rs (hunk 3 above)
Comment (dmitry-timofeev) [reply src/wrapper/jnienv.rs:1766]: That's correct: arrays are object references in Java, and are represented in JValue enum as JValue::Object (there is no JValue::Array). The C JNI API accepts them as an argument to `SetStaticObjectField`

**Review by dmitry-timofeev — APPROVED** [review]
Looks good. Regarding testing: I do not happen to know a class in the standard libary wit h a public static non-final field that we can set. The JNI methods, however, allow getting field IDs of private fields, but I do not think it's good to make a test depending on a private field in a std. library class. On top of that, VMs may be free to forbid such access to platform classes, but I haven't checked that.

Ideally, we could have a separate Java module with some classes for testing, but we are not there yet.

**Pull request merged**